editor/import/3d/post_import_plugin_skeleton_rest_fixer.cpp

```diff
@@ -447,6 +448,48 @@ void PostImportPluginSkeletonRestFixer::internal_process(InternalImportCategory
 				old_skeleton_global_rest.push_back(src_skeleton->get_bone_global_rest(i));
 			}
 
+			bool skip_unmapped_ends = bool(p_options["retarget/rest_fixer/skip_unmapped_ends"]);
+
+			// Scan hierarchy and populate a whitelist of unmapped bones without mapped descendants.
+			Vector<int> keep_bone_rest;
+			if (skip_unmapped_ends) {
+				Vector<int> bones_to_process = src_skeleton->get_parentless_bones();
+				while (bones_to_process.size() > 0) {
+					int src_idx = bones_to_process[0];
+					bones_to_process.erase(src_idx);
+					Vector<int> src_children = src_skeleton->get_bone_children(src_idx);
+					for (int i = 0; i < src_children.size(); i++) {
+						bones_to_process.push_back(src_children[i]);
```

Comment:
```suggestion
					for (const int &src_child : src_children) {
						bones_to_process.push_back(src_child);
```
You can use for-each syntax here